src/format/formatting.rs

```diff
@@ -97,7 +97,38 @@ impl<'a, I: Iterator<Item = B> + Clone, B: Borrow<Item<'a>>> DelayedFormat<I> {
         DelayedFormat { date, time, off: Some(name_and_diff), items, locale }
     }
 
-    fn format(&self, w: &mut impl Write) -> fmt::Result {
+    /// Formats `DelayedFormat` into a `core::fmt::Write` instance.
+    /// # Errors
+    /// This function returns a `core::fmt::Error` if formatting into the `core::fmt::Write` instance fails.
+    ///
+    /// # Example
+    /// ### Writing to a String
+    /// ```
+    /// let dt = chrono::DateTime::from_timestamp(1643723400, 123456789).unwrap();
+    /// let df = dt.format("%Y-%m-%d %H:%M:%S%.9f");
+    /// let mut buffer = String::new();
+    /// let _ = df.write_to(&mut buffer);
+    /// ```
+    /// ### Writing to a Vec
+    /// ```
+    /// // wrapper to allow reuse of the existing string based
+    /// // writers
+    /// struct IoWriter<W: std::io::Write> {
+    ///     writer: W,
+    /// }
+    /// impl<W: std::io::Write> std::fmt::Write for IoWriter<W> {
+    ///     #[inline]
+    ///     fn write_str(&mut self, s: &str) -> std::fmt::Result {
+    ///         self.writer.write_all(s.as_bytes()).map_err(|_| std::fmt::Error)
+    ///     }
+    /// }
+    /// let dt = chrono::DateTime::from_timestamp(1643723400, 123456789).unwrap();
+    /// let df = dt.format("%Y-%m-%d %H:%M:%S%.9f");
+    /// let w: Vec<u8> = Vec::new();
+    /// let mut writer = IoWriter { writer: w };
+    /// let _ = df.write_to(&mut writer);
+    /// ```
```

Comment:
Yes, also not interested in having the `Write` adapter in the examples.

Reply:
done